smoke_test/tests/authenticated/sam_user_info_tests.py

```diff
@@ -0,0 +1,20 @@
+import json
+
+from ..sam_smoke_test_case import SamSmokeTestCase
+
+
+class SamUserInfoTests(SamSmokeTestCase):
+    @staticmethod
+    def user_info_url() -> str:
+        return SamSmokeTestCase.build_sam_url("/register/user/v2/self/info")
+
+    def test_status_code_is_200(self):
+        response = SamSmokeTestCase.call_sam(self.user_info_url(), SamSmokeTestCase.USER_TOKEN)
+        self.assertEqual(response.status_code, 200, f"User info HTTP Status is not 200: {response.text}")
+
+    def test_content_contains_correct_keys(self):
+        response = SamSmokeTestCase.call_sam(self.user_info_url(), SamSmokeTestCase.USER_TOKEN)
+        user_info = json.loads(response.text)
+        self.assertListEqual(list(user_info.keys()),
+                             ["adminEnabled", "enabled", "userEmail", "userSubjectId"],
+                             "User Info object does not match expected fields")
```

Comment:
is it worth maybe creating/deleting a user, something like that?

another thought is we create a test user and auth them successfully for a test

Reply:
We don't have APIs for deleting users.

I like the idea of adding a test for creating a user but feel conflicted about it. I think the tests were intended to cover common test scenarios. Creating a user could be useful but it's typically not something we test. But it is an essential part of the system and is a bit annoying to test, which makes it a good candidate for automating. Maybe we start with these tests for now and add that later?

Reply:
My take:

1.  I don't want the smoke tests to affect the state of the data or the service in any way.  This will avoid conflicts of multiple people running these tests at the same time or the potential of a test failing to clean up after itself and leaving the database in a partially cleaned up state, etc.  Additionally, by not worrying about modifying state in any way, these tests become much more portable.  They require nothing to run (beyond python) and can target any environment.  The only requirement for the test runner is that they _may optionally_ choose to provide a user access token.  
2.  Smoke tests are not end-to-end (E2E) or full functional tests of the application.  These are intended to be small, quick idempotent tests that just "sniff" the app to see that it generally is up and can perform the most basic functionality.  If we want to write more complex Sam E2E tests, we can do that, but I would prefer those to be a separate suite/script because these will be much more involved and how/when we run them will be very different.

Reply:
Agreed sounds good.